tests/serialization/test_modules.py

```diff
@@ -0,0 +1,26 @@
+"""Tests for citrine.informatics.design_spaces serialization."""
+import uuid
+
+import pytest
+
+from citrine.informatics.modules import Module
+from citrine.informatics.design_spaces import ProductDesignSpace
+from citrine.informatics.predictors import SimpleMLPredictor
+from citrine.informatics.processors import GridProcessor
+
+
+def test_polymorphic_design_space_deserialization(valid_product_design_space_data):
+    """Ensure that a deserialized ProductDesignSpace looks sane."""
+    module: Module = Module.build(valid_product_design_space_data)
```

Comment:
I think we can defer this to a later PR, but I think the answer to getting around the circular dependencies is to not have `Module` responsible for building this.  If you have some other `Builder` class that knows how to translate json into a concrete class, then `Module` doesn't need to know about any subclasses.  That was my only point previously.

Reply:
Sure, but that's completely changing the pattern that the entire library is based on.